### DISCUSSION TITLE
Validating inputs after triggering them

### DISCUSSION THREAD

**Discussion by vmichailidisdev** [react-hook-form/react-hook-form#7962] [Q&A] [answered]
Hello,
I am triggering some required inputs and when I am getting errors and changing those inputs they won't automatically remove the error.
I want to imitate the functionality of handleSubmit:
Submit the form -> I have errors in some inputs -> Changing them -> automatically the validation procs.

Thank you!

**Comment by fahadsohail482** [comment, marked as answer]
@vmichailidisdev Please use the key mode:"onChange" in useForm to validate on each input's value change.
This will automatically remove the error when your input has some value.
` const methods = useForm({  mode: "onChange" });`
This will solve your problem.

**Comment by tidusia** [comment]
@vmichailidisdev I don't know if it's a good practice but here is a solution I found that does not require you to change the form `mode` : 

```ts
const triggerAndAttachOnChangeEvents = async () => {
    // Hack to attach onChange events to inputs so they can revalidate after trigger
    await handleSubmit(() => null)();
    return await formUtils.trigger();
  };
```
